### PR TITLE
upgrade from Bullet-2.83 to Bullet-2.88

### PR DESCRIPTION
**Pull request by AndrewMeadows**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9540/HighFidelity-Beta-PR13109-080b5e7cfe27dede78abd79d5fc01de1190bebc0.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9540/HighFidelity-Beta-Interface-PR13109-080b5e7cfe27dede78abd79d5fc01de1190bebc0.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9540/HighFidelity-Beta-PR13109-080b5e7cfe27dede78abd79d5fc01de1190bebc0.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9540/HighFidelity-Beta-Interface-PR13109-080b5e7cfe27dede78abd79d5fc01de1190bebc0.exe

**Comment by sethalves** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9547/HighFidelity-Beta-PR13109-080b5e7cfe27dede78abd79d5fc01de1190bebc0.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9547/HighFidelity-Beta-Interface-PR13109-080b5e7cfe27dede78abd79d5fc01de1190bebc0.exe

**Comment by AndrewMeadows** [comment]
This PR is missing an update to the android pre-built dependencies.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9587/HighFidelity-Beta-PR13109-56c08050248c263018b2ad513262fab1fe1c58c2.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9587/HighFidelity-Beta-Interface-PR13109-56c08050248c263018b2ad513262fab1fe1c58c2.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9587/HighFidelity-Beta-PR13109-56c08050248c263018b2ad513262fab1fe1c58c2.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9587/HighFidelity-Beta-Interface-PR13109-56c08050248c263018b2ad513262fab1fe1c58c2.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9609/HighFidelity-Beta-PR13109-c91879d49e34a17f5d6cc6234f4017eb555d64e2.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9609/HighFidelity-Beta-Interface-PR13109-c91879d49e34a17f5d6cc6234f4017eb555d64e2.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9609/HighFidelity-Beta-PR13109-c91879d49e34a17f5d6cc6234f4017eb555d64e2.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9609/HighFidelity-Beta-Interface-PR13109-c91879d49e34a17f5d6cc6234f4017eb555d64e2.exe

**Comment by AndrewMeadows** [comment]
The motivation for this PR is to fix some "address sanitizer" warnings that were deep in Bullet:
https://highfidelity.fogbugz.com/f/cases/14676/PhysicsEngine-updateContactMap-asan-warning
https://highfidelity.fogbugz.com/f/cases/14675/CharacterController-asan-warning

File: android/build.gradle [hunk 1]
```diff
@@ -608,4 +609,4 @@ task testElf (dependsOn: 'externalNativeBuildDebug') {
         }
     }
 }
-*/
\ No newline at end of file
+*/
```
Comment (AndrewMeadows) [inline android/build.gradle]: Most of these changes to `android/build.gradle` are a temporary HACK to allow us to use different S3 buckets for the various android dependencies.  In the near future @SamGondelman will move all android dependencies properly over to the `hifi-public.s3.amazonaws.com/dependencies/android/` bucket.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9647/app_13109_PR_e967585882630b16a4b795c3acfa51b0010a4660.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9647/HighFidelity-Beta-PR13109-e967585882630b16a4b795c3acfa51b0010a4660.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9647/HighFidelity-Beta-Interface-PR13109-e967585882630b16a4b795c3acfa51b0010a4660.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9647/HighFidelity-Beta-PR13109-e967585882630b16a4b795c3acfa51b0010a4660.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9647/HighFidelity-Beta-Interface-PR13109-e967585882630b16a4b795c3acfa51b0010a4660.exe

**Comment by AndrewMeadows** [comment]
There is an effective protocol change in this PR because the number of bits in the `collisionMask` property has increased.  I need to either bump the protocol version, or figure out how to use only 8 bits of `collisionMask` despite 16 being available.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9659/HighFidelity-Beta-PR13109-44f1a53ffa3a4d1a1f689429ce3cf97f2ec1978f.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9659/HighFidelity-Beta-Interface-PR13109-44f1a53ffa3a4d1a1f689429ce3cf97f2ec1978f.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9659/HighFidelity-Beta-PR13109-44f1a53ffa3a4d1a1f689429ce3cf97f2ec1978f.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9659/HighFidelity-Beta-Interface-PR13109-44f1a53ffa3a4d1a1f689429ce3cf97f2ec1978f.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10044/HighFidelity-Beta-PR13109-fc85d0a67ae31f75f79d9995aa84932c59afda29.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10044/HighFidelity-Beta-PR13109-fc85d0a67ae31f75f79d9995aa84932c59afda29.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10044/HighFidelity-Beta-Interface-PR13109-fc85d0a67ae31f75f79d9995aa84932c59afda29.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10044/HighFidelity-Beta-PR13109-fc85d0a67ae31f75f79d9995aa84932c59afda29.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10044/HighFidelity-Beta-Interface-PR13109-fc85d0a67ae31f75f79d9995aa84932c59afda29.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10061/HighFidelity-Beta-PR13109-c5da4a9c8026797c805382b55a66a07e86e6418f.apk

**Comment by qlimnavarro** [comment]
Pass @conklin94122 
Numbers are lower in this PR than in master, Master I would see spikes go up to about 1.9ms and in this PR it would not go above 1.3! Walking up slopes, flying and grabbing all look as expected.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10061/HighFidelity-Beta-PR13109-c5da4a9c8026797c805382b55a66a07e86e6418f.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10061/HighFidelity-Beta-Interface-PR13109-c5da4a9c8026797c805382b55a66a07e86e6418f.exe

**Comment by samcake** [comment]
Waiting on Android test pass before removing the dnm

**Comment by AndrewMeadows** [comment]
It has been reported the android build works fine, so I removed the **DO_NOT_MERGE** label.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10087/HighFidelity-Beta-PR13109-288efb1f93fe930d16aed606eb09804d7f7d1a56.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10087/HighFidelity-Beta-Interface-PR13109-288efb1f93fe930d16aed606eb09804d7f7d1a56.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10087/HighFidelity-Beta-PR13109-288efb1f93fe930d16aed606eb09804d7f7d1a56.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10087/HighFidelity-Beta-Interface-PR13109-288efb1f93fe930d16aed606eb09804d7f7d1a56.exe

**Pull request merged**